src/logic/useRealtimeVehicleData.ts

```diff
@@ -0,0 +1,149 @@
+import { useQuery, useSubscription } from '@apollo/client'
+import type { OnSubscriptionDataOptions } from '@apollo/client'
+
+import { useCallback, useEffect, useState } from 'react'
+
+import { Filter } from '../services/realtimeVehicles/types/filter'
+import { RealtimeVehicle } from '../services/realtimeVehicles/types/realtimeVehicle'
+
+import {
+    VEHICLES_QUERY,
+    VEHICLE_UPDATES_SUBSCRIPTION,
+} from '../services/realtimeVehicles/graphql'
+
+import {
+    SWEEP_INTERVAL_MS,
+    DEFAULT_FETCH_POLICY,
+    BUFFER_SIZE,
+    BUFFER_TIME,
+} from '../constants'
+
+import { useSettingsContext } from '../settings'
+
+import { useStopPlacesWithLines } from './useStopPlacesWithLines'
+
+import useVehicleReducer, { ActionType } from './useRealtimeVehicleReducer'
+
+interface Return {
+    realtimeVehicles: RealtimeVehicle[] | undefined
+    allLinesWithRealtimeData: string[] | undefined
+}
+interface SubscriptionData {
+    vehicleUpdates: RealtimeVehicle[]
+}
+interface QueryData {
+    vehicles: RealtimeVehicle[]
+}
+
+/**
+ * Hook to query and subscribe to remote vehicle data
+ */
+export default function useVehicleData(filter?: Filter): Return {
+    const [state, dispatch] = useVehicleReducer()
+    const { uniqueLines } = useStopPlacesWithLines()
+    const [settings] = useSettingsContext()
+    const { hiddenRealtimeDataLineRefs } = settings || {}
+    const [realtimeVehicles, setRealtimeVehicles] = useState<
+        RealtimeVehicle[] | undefined
+    >(undefined)
+    const [allLinesWithRealtimeData, setAllLinesWithRealtimeData] = useState<
+        string[] | undefined
+    >(undefined)
+
+    const filterVehiclesByLineRefs = useCallback(
+        (vehiclesUpdates: RealtimeVehicle[] | undefined) => {
+            if (!(vehiclesUpdates && uniqueLines)) {
+                return undefined
+            }
+
+            const filteredUpdates = vehiclesUpdates.filter(
+                (vehicle) =>
+                    uniqueLines
+                        .map((line) => line.id)
+                        .includes(vehicle.line.lineRef) &&
+                    hiddenRealtimeDataLineRefs &&
+                    !hiddenRealtimeDataLineRefs?.includes(vehicle.line.lineRef),
+            )
+
+            return filteredUpdates.length > 0 ? filteredUpdates : undefined
+        },
+        [uniqueLines, hiddenRealtimeDataLineRefs],
+    )
+
+    const handleQueryData = useCallback(
+        ({ vehicles }: QueryData) => {
+            setAllLinesWithRealtimeData(
+                new Array(
```

Comment:
I staden for `new Array` trur eg du kan berre bruke vanlig `[]`?
```
setAllLinesWithRealtimeData(
  [
    ...new Set(...)
  ]
)
```

Reply:
Jepp, det funket fint:)